### PR TITLE
`HeaderUtils`: fix problems with missing space after semicolon

### PR DESCRIPTION
**Pull request by idelpivnitskiy**
Motivation:

https://github.com/netty/netty/pull/12833 adds additional 3 commits to fix problems with missing space after semicolon and adds more tests.

Modifications:

- Port changes made in the last 3 commits of Netty PR;
- Remove unused internal methods;
- Consistently throw `cookie is not allowed to end with ;` from anywhere we parse cookies;
- Include `DefaultHttpSetCookiesTest` and `LegacyCookieParsingTest` for `testProps`;
- Apply all settings we have to `test` task to `testProps` too: timeout, logging, args, etc;

Result:

Correct parsing of cookies without SP after a semicolon.

### CODE REVIEW

**Review by chrisvest — APPROVED** [review]

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java [hunk 1]
```diff
@@ -557,14 +557,33 @@ protected final void initNext(CharSequence cookieHeaderValue) {
...
+            }
+            HttpCookiePair next = DefaultHttpCookiePair.parseCookiePair(
+                    cookieHeaderValue, nextNextStart, nameLength, semiIndex);
             if (semiIndex > 0) {
                 if (cookieHeaderValue.length() - 2 <= semiIndex) {
-                    advanceCookieHeaderValue();
-                    nextNextStart = 0;
-                } else {
-                    // skip 2 characters "; " (see https://tools.ietf.org/html/rfc6265#section-4.2.1)
+                    if (cookieParsingStrictRfc6265()) {
```
Comment (Scottmitch) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java:570]: when you rebase this method no longer exists. just use the static final variable instead.

File: servicetalk-http-api/src/test/java/io/servicetalk/http/api/LegacyCookieParsingTest.java [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+ * We recreate those tests here, to verify that we're still compatible with those formats.
+ */
+class LegacyCookieParsingTest {
+
+    private static final HttpHeadersFactory NO_VALIDATION = new DefaultHttpHeadersFactory(false, false, false);
+    HttpHeaders headers;
+
+    @BeforeEach
+    void createHeaders() {
+        headers = NO_VALIDATION.newHeaders();
```
Comment (Scottmitch) [inline servicetalk-http-api/src/test/java/io/servicetalk/http/api/LegacyCookieParsingTest.java:71]: will these tests fail if strict validation is used? if so consider adding an `assumeFalse(<strict_validation_systemp_prop>)`

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java [hunk 3]
```diff
@@ -557,14 +557,33 @@ protected final void initNext(CharSequence cookieHeaderValue) {
          */
         private HttpCookiePair findNext(CharSequence cookieHeaderValue) {
             int semiIndex = nextCookieDelimiter(cookieHeaderValue, nextNextStart);
-            HttpCookiePair next = DefaultHttpCookiePair.parseCookiePair(cookieHeaderValue, nextNextStart, semiIndex);
+            int nameLength = indexOf(cookieHeaderValue, '=', nextNextStart) - nextNextStart;
```
Comment (Scottmitch) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java:560]: the previous `parseCookiePair` method is no longer used, remove it?

should the `nameLength` validation be done in the `parseCookiePair` method?

File: servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java [hunk 4]
```diff
@@ -557,14 +557,33 @@ protected final void initNext(CharSequence cookieHeaderValue) {
...
+                    cookieHeaderValue, nextNextStart, nameLength, semiIndex);
             if (semiIndex > 0) {
                 if (cookieHeaderValue.length() - 2 <= semiIndex) {
-                    advanceCookieHeaderValue();
-                    nextNextStart = 0;
-                } else {
-                    // skip 2 characters "; " (see https://tools.ietf.org/html/rfc6265#section-4.2.1)
+                    if (cookieParsingStrictRfc6265()) {
+                        throw new IllegalArgumentException("cookie '" + next.name() +
+                                "': cookie is not allowed to end with ;");
```
Comment (Scottmitch) [inline servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java:572]: `H2PriorKnowledgeFeatureParityTest.cookiesRoundTrip` has a `endsWithSemi` variable now which isn't picked up as invalid. can we make the parsing/validation consistent?

**Pull request merged**